### ISSUE TITLE
Copy pasting cells with formulae between files does not store the formula in the new file

### ISSUE THREAD

**Issue by IsakNaslundBh** [BHoM/Excel_UI#249]
#### Description:

If one copy pastes cells with BHoMFormula to a sheet without the same BHoM formula in it, for example to a new sheet, it runs fine whilst still open. Saving the new sheet and closing it though, the formula does not load up, but gives a `#NAME` value in the cell. Adding the formula used to another cell of the sheet, the cell starts working again.

Seems as copy pasted cells containing BHoM calls do not get correctly serialised to the sheet.

#### Steps to reproduce:

- Open a fresh sheet
- Add some BHoM method call
- Open another new sheet
- Copy paste the cells containing BHoM calls to the new sheet
- Save and close the sheet containing the pasted cells
- Re-open the saved and closed sheet

#### Expected behaviour:

For copy pasted BHoM formulations to be saved

#### Test file(s):

No needed/possible

**Comment by IsakNaslundBh** [comment]
Noting as a comment here. The issue experienced here can be fixed by adding the method(s) missing via either the ctrl+shift+b or the ribbon. This can be done before the sheet is first closed, or after it is re-opened and the issue is experienced. After this has been done, the issue is fixed for that sheet and opens as expected.

**Comment by adecler** [comment]
Once we have registered a formula with Excel through the global menu or the ribbon, we surrender the control of it to Excel. This means that a copy-paste of that formula is purely Excel's responsibility. The side effect is that we do no save the formula in the new hidden spreadsheet if it is copies between files. 

We do, however, have a hook when the formula is ran. So we could check that is is in the hidden sheet every time is is executed. The way the check is executed right now is very safe but also quite slow (we go through the hidden sheet to see if that method is there already). I don't want to touch that part of the code this close to the beta release so I will fix that at the beginning of 4.0.

**Comment by adecler** [comment]
@IsakNaslundBh , I just came back to this.
Can you confirm that, by `sheet`, you mean `file` ?

I have tested your steps above between sheets in the same file just to make sure and it works perfectly as expected.

**Comment by IsakNaslundBh** [comment]
Yes, it only had issues between different files. Sorry, got the terminology wrong!

Inside the same file all works fine when copy pasting between the sheets, but between sheets in different files it does not.

**Comment by adecler** [comment]
OK, thanks @IsakNaslundBh . Moving this to feature then. Although I am not sure how we can make this work as the BHoM is not involved in the copy-paste process at all.

**Comment by IsakNaslundBh** [comment]
Had this problem again the other day, and Seem to be related to the way the methods used are stored, on the hidden BHoM sheets. That seem to be added to the file/workbook that was opened first, and then just added to, even if you add BHoM methods in a separate open file. This essentially means that if you want to work with BHoM in excel, you need to make sure to only have a single Excel file open.

Wonder if this can be fixed by storing the used methods elsewhere, and maybe not on a hidden sheet, but stashed away in some other field of the XML document or similar.
